pkg/operator/ceph/csi/controller.go

```diff
@@ -118,22 +117,32 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to list ceph clusters")
 	}
 
-	// // Do not nothing if no ceph cluster is present
+	// Do nothing if no ceph cluster is present
 	if len(cephClusters.Items) == 0 {
 		logger.Debug("no ceph cluster found not deploying ceph csi driver")
 		return reconcile.Result{}, nil
-	} else {
-		for _, cluster := range cephClusters.Items {
-			if !cluster.DeletionTimestamp.IsZero() {
-				logger.Debug("ceph cluster is being deleting, no need to reconcile the csi driver")
-				return reconcile.Result{}, nil
-			}
-
-			if !cluster.Spec.External.Enable && cluster.Spec.CleanupPolicy.HasDataDirCleanPolicy() {
-				logger.Debug("ceph cluster has cleanup policy, the cluster will soon go away, no need to reconcile the csi driver")
-				return reconcile.Result{}, nil
-			}
+	}
+
+	var cluster cephv1.CephCluster
+	for _, tmpCluster := range cephClusters.Items {
+		if tmpCluster.ObjectMeta.Namespace == request.NamespacedName.Namespace {
+			cluster = tmpCluster
+			break
+		}
+	}
+
+	if !cluster.DeletionTimestamp.IsZero() {
```

Comment:
Instead of triggering the teardown from the deletion timestamp, how about from within the `stopDrivers()` method when the entire csi driver is also stopped? There is just one daemonset to remove from the same namespace as the operator, right? If so, then no need to trigger this from the cluster CR.